.github/workflows/shell-diff-lint.yml

```diff
@@ -0,0 +1,24 @@
+---
+name: Lint Shell issues
+
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  shell-lint-job:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Differential ShellCheck
+        uses: redhat-plumbers-in-action/differential-shellcheck@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
even though, why do you need this?

Reply:
It is used for uploading SARIF results to GitHub - [doc](https://github.com/redhat-plumbers-in-action/differential-shellcheck#token).